lib/Models/RegionParameter.js

```diff
@@ -96,7 +96,11 @@ RegionParameter.prototype.findRegionByID = function(regionID) {
     }
 
     return regionProvider.regions.filter(function(region) {
-        return region.id === regionID;
+        if (typeof region.id === 'string' && typeof regionID === 'string') {
```

Comment:
Why is this change necessary?  I think it's ok for region IDs to be case sensitive.  Or, put another way, the replacements system should take care of any adjustments to the ID that are necessary.